packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -1,21 +1,61 @@
 /*
- * Copyright (c) 2022, Salesforce, Inc.
+ * Copyright (c) 2023, Salesforce, Inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import {ShopperBaskets} from 'commerce-sdk-isomorphic'
-import {ShopperContexts} from 'commerce-sdk-isomorphic'
-import {ShopperCustomers} from 'commerce-sdk-isomorphic'
-import {ShopperDiscoverySearch} from 'commerce-sdk-isomorphic'
-import {ShopperExperience} from 'commerce-sdk-isomorphic'
-import {ShopperGiftCertificates} from 'commerce-sdk-isomorphic'
-import {ShopperLogin} from 'commerce-sdk-isomorphic'
-import {ShopperOrders} from 'commerce-sdk-isomorphic'
-import {ShopperProducts} from 'commerce-sdk-isomorphic'
-import {ShopperPromotions} from 'commerce-sdk-isomorphic'
-import {ShopperSearch} from 'commerce-sdk-isomorphic'
-import {QueryKey, QueryFunctionContext} from '@tanstack/react-query'
+import {InvalidateQueryFilters, QueryFilters, Updater, UseQueryOptions} from '@tanstack/react-query'
+import {
+    ShopperBaskets,
+    ShopperContexts,
+    ShopperCustomers,
+    ShopperExperience,
+    ShopperGiftCertificates,
+    ShopperLogin,
+    ShopperOrders,
+    ShopperProducts,
+    ShopperPromotions,
+    ShopperSearch
+} from 'commerce-sdk-isomorphic'
+
+// --- GENERAL UTILITIES --- //
+
+/** Makes a type easier to read. */
+export type Prettify<T extends object> = NonNullable<Pick<T, keyof T>>
+
+/**
+ * Marks the given keys as required.
+ * WARNING: Does not work if T has an index signature.
+ */
+export type RequireKeys<T, K extends keyof T> = Prettify<T & Required<Pick<T, K>>>
+
+/** Removes keys whose value is `never`. */
+type RemoveNeverValues<T> = {
+    [K in keyof T as T[K] extends never ? never : K]: T[K]
+}
+
+/** Change string index type to `never`. */
+type StringIndexNever<T> = {
```

Comment:
This makes it more readable
```suggestion
type StringIndexToNever<T> = {
```